src/Core/src/Credentials/ContainerProvider.php

```diff
@@ -77,4 +100,78 @@ public function getCredentials(Configuration $configuration): ?Credentials
             Credentials::adjustExpireDate(new \DateTimeImmutable($result['Expiration']), $date)
         );
     }
+
+    /**
+     * Checks if the provided IP address is a loopback address.
+     *
+     * @param string $host the host address to check
+     *
+     * @return bool true if the IP is a loopback address, false otherwise
+     */
+    public static function isLoopBackAddress(string $host)
```

Comment:
no need to be static, nor to be public. 
lets avoid exposing ( = maintaining) new methods